### ISSUE TITLE
Ability to run go-pry outside of main package

### ISSUE THREAD

**Issue by catherinetcai** [d4l3k/go-pry#19]
I'm wondering if it's in the scope of go-pry to eventually allow it to run in non-main packages. The way I currently have my Go project set up is to have a main package that sources certain functionalities from a ton of subpackages.

**Comment by d4l3k** [comment]
Yeah, it'd be definitely nice to support non-main packages. Since go-pry
works by injecting code you'd either have to specify the package or do a
very expensive scan of the entire dependency tree.

**Comment by d4l3k** [comment]
Sub packages within the same directory should work however

**Comment by tomaswitek** [comment]
Hi @d4l3k thx for you great project.
I am comming from Ruby environment so I am very happy to see a REPL for go 🚀 
Just a question is there already a way how I could specify non-main package?

**Comment by d4l3k** [comment]
There's currently no way, but it jwould make a lot of sense to add a new flag to specify extra packages to pry into. Like `go-pry run -pkgs="pkg1,pkg2" foo.go`
